### PR TITLE
Add support for 1.4: vulnerabilities

### PR DESCRIPTION
**Pull request by tokcum**
Hi @amy-keibler,

this is my first draft to add support for 1.4 to cyclonedx-bom. All unit & integration tests run fine. However, there are still some todos to fully comply with 1.4.

Anyway, I would like to raise this pull request to align next steps. I recently joined the CycloneDX Slack channel. Let me know what you think. 

Thanks.

### CODE REVIEW

**Comment by amy-keibler** [comment]
Thanks for opening this PR! It will take me a bit to go over it in detail (especially the `insta` test expectations), but a few notes before that:

- If you rebase off the latest `main`, you should get the `clippy` CI check down to just the ones relevant to your PR
- We use git's signoff command to handle copyright tracking (to indicate that you are the author of the code for copyright purposes), so you'll have to `git commit --amend --signoff` this commit and sign follow-up commits to get the DCO CI check to pass

**Comment by tokcum** [comment]
Thank you, @amy-keibler. Sorry, I missed the clippy tests and I'll also take care of the signed-off commit.

File: cyclonedx-bom/src/specs/v1_4/component.rs [hunk 1]
```diff
@@ -0,0 +1,1468 @@
...
+    ) -> Result<Self, XmlReadError>
+    where
+        Self: Sized,
+    {
+        let tag_id = attribute_or_error(element_name, attributes, TAG_ID_ATTR)?;
+        let name = attribute_or_error(element_name, attributes, NAME_ATTR)?;
+        let version = optional_attribute(attributes, VERSION_ATTR);
+        let tag_version =
+            if let Some(tag_version) = optional_attribute(attributes, TAG_VERSION_ATTR) {
+                let tag_version = u32::from_xml_value(TAG_VERSION_ATTR, &tag_version)?;
```
[the remark this answers is not in the document]
Comment (tokcum) [reply cyclonedx-bom/src/specs/v1_4/component.rs:672]: Hm, not sure why clippy fails again. This issue is not found by my local clippy.

**Comment by tokcum** [comment]
Proof of concept: models::component.version is now an Option, while component.version is mandatory in spec::v1_3. Using TryFrom instead of From for component.version was pretty straight forward. I think this works well. However, we have to think about Error types, how we collect and propagate them.

This PoC is not very elegant yet. Many things could and should be improved.

**Pull request closed without merging**